code/modules/mob/living/simple_animal/bot/bot_announcement.dm

```diff
@@ -0,0 +1,44 @@
+/// Say something and play a corresponding sound effect
+/datum/action/cooldown/bot_announcement
+	name = "Make automated announcement"
+	desc = "Play a prerecorded message for the benefit of those around you."
+	background_icon_state = "bg_tech_blue"
+	overlay_icon_state = "bg_tech_blue_border"
+	button_icon = 'icons/mob/actions/actions_AI.dmi'
+	button_icon_state = "intercom"
+	cooldown_time = 10 SECONDS
+	melee_cooldown_time = 0 SECONDS
+	/// List of strings to sound effects corresponding to automated messages we can play
+	var/list/automated_announcements
+
+/datum/action/cooldown/bot_announcement/New(Target, original, list/automated_announcements)
+	src.automated_announcements = automated_announcements
+	return ..()
+
+/datum/action/cooldown/bot_announcement/IsAvailable(feedback)
+	. = ..()
+	if (!.)
+		return
+	if (!isbot(owner))
+		if (feedback)
+			owner.balloon_alert(owner, "no announcement system!")
+		return FALSE
+	if (!length(automated_announcements))
+		if (feedback)
+			owner.balloon_alert(owner, "no valid announcements!")
+		return FALSE
+	return TRUE
+
+/datum/action/cooldown/bot_announcement/Trigger(trigger_flags, atom/target)
```

Comment:
Cooldown actions shouldn't override `Trigger` for this purpose, they have `Activate`

(In addition, not checking the return value of parent means all the code in `IsAvailable` does nothing)